homeassistant/components/websocket_api/http.py

```diff
@@ -161,11 +163,31 @@ async def _writer(self) -> None:
                     messages.append(process if isinstance(process, str) else process())
                     messages_remaining -= 1
 
-                joined_messages = ",".join(messages)
-                coalesced_messages = f"[{joined_messages}]"
-                if debug_enabled:
-                    debug("%s: Sending %s", self.description, coalesced_messages)
-                await send_str(coalesced_messages)
+                coalesced_messages = f"[{','.join(messages)}]"
+                # We only check the length of the coalesced messages after
+                # combining them to avoid the overhead of checking the length
+                # of each message individually, and since the majority of
+                # the time we will never hit the max message size limit this
+                # is a good tradeoff.
+                if len(coalesced_messages) < max_message_size:
+                    if debug_enabled:
+                        debug("%s: Sending %s", self.description, coalesced_messages)
+                    await send_str(coalesced_messages)
+                    continue
+
+                logger.warning(
+                    "%s: Coalesced messages exceeded maximum size (%s/%s), this usually "
+                    "indicates an integration is sending too much data, one of the "
+                    "registries is too large, or the system cannot "
+                    "keep up; trying to send individually",
+                    self.description,
+                    len(coalesced_messages),
+                    max_message_size,
+                )
```

Comment:
I made this a warning since we never except this to happen when the system is able to keep up. I was worried we would have no indication that something is overloaded behind the scenes if we don't log something here.